cognite/client/data_classes/agents/chat.py

```diff
@@ -85,6 +86,194 @@ def _load_content(cls, data: dict[str, Any]) -> UnknownContent:
+}
+
+
+@dataclass
+class ActionCall(CogniteObject, ABC):
+    """Base class for action calls requested by the agent."""
+
+    _type: ClassVar[str]
+    action_id: str
+
+    @classmethod
+    def _load(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> ActionCall:
+        """Dispatch to the correct concrete action call class based on `type`."""
+        action_type = data.get("type", "")
+        action_class = _ACTION_CALL_CLS_BY_TYPE.get(action_type, UnknownActionCall)
+        return action_class._load_call(data, cognite_client)
+
+    @abstractmethod
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        """Dump the action call to a dictionary."""
+        ...
+
+    @classmethod
+    @abstractmethod
+    def _load_call(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> ActionCall:
+        """Create a concrete action call instance from raw data."""
+        ...
+
+
+@dataclass
+class ClientToolCall(ActionCall):
+    """A client tool call requested by the agent.
+
+    Args:
+        action_id (str): The unique identifier for this action call.
+        name (str): The name of the client tool being called.
+        arguments (dict[str, Any]): The parsed arguments for the tool call.
+    """
+
+    _type: ClassVar[str] = "clientTool"
+    action_id: str
+    name: str
+    arguments: dict[str, Any]
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {
+            "type": self._type,
+            "actionId" if camel_case else "action_id": self.action_id,
+            "clientTool": {
+                "name": self.name,
+                "arguments": json.dumps(self.arguments),
+            },
+        }
+
+    @classmethod
+    def _load_call(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> ClientToolCall:
+        client_tool = data["clientTool"]
+        arguments_str = client_tool["arguments"]
+        return cls(
+            action_id=data["actionId"],
+            name=client_tool["name"],
+            arguments=json.loads(arguments_str),
+        )
+
+
+@dataclass
+class UnknownActionCall(ActionCall):
+    """Unknown action call type for forward compatibility.
+
+    Args:
+        action_id (str): The unique identifier for this action call.
+        type (str): The action call type.
+        data (dict[str, Any]): The raw action call data.
+    """
+
+    action_id: str
+    type: str
+    data: dict[str, Any] = field(default_factory=dict)
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        result = self.data.copy()
+        result["type"] = self.type
+        result["actionId" if camel_case else "action_id"] = self.action_id
+        return result
+
+    @classmethod
+    def _load_call(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> UnknownActionCall:
+        action_type = data.get("type", "")
+        action_id = data.get("actionId", "")
```

Comment:
shouldn't type and actionID always be present?
```suggestion
        action_type = data["type"]
        action_id = data["actionId"]
```